ui/apps/platform/src/Components/CompoundSearchFilter/components/SearchFilterAutocomplete.tsx

```diff
@@ -0,0 +1,288 @@
+
+    const onInputKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        const enabledMenuItems = selectOptions.filter((option) => !option.isDisabled);
+        const [firstMenuItem] = enabledMenuItems;
+        const focusedItem = focusedItemIndex ? enabledMenuItems[focusedItemIndex] : firstMenuItem;
+
+        switch (event.key) {
+            // Select the first available option
+            case 'Enter':
+                if (isOpen && focusedItem.value !== 'no results') {
+                    onChange(String(focusedItem.children));
+                    setFilterValue('');
+                }
+
+                setIsOpen((prevIsOpen) => !prevIsOpen);
+                setFocusedItemIndex(null);
+                setActiveItem(null);
+
+                break;
+            case 'Tab':
+            case 'Escape':
+                setIsOpen(false);
+                setActiveItem(null);
+                break;
+            case 'ArrowUp':
+            case 'ArrowDown':
+                event.preventDefault();
+                handleMenuArrowKeys(event.key);
+                break;
+            default:
+        }
+    };
+
+    const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+            ref={toggleRef}
+            variant="typeahead"
+            onClick={onToggleClick}
+            isExpanded={isOpen}
+            isFullWidth
+            aria-labelledby={`${textLabel} menu toggle`}
+        >
+            <TextInputGroup isPlain>
+                <TextInputGroupMain
+                    value={value}
+                    onClick={onToggleClick}
+                    onChange={onTextInputChange}
+                    onKeyDown={onInputKeyDown}
+                    id="typeahead-select-input"
+                    autoComplete="off"
+                    innerRef={textInputRef}
+                    placeholder={textLabel}
+                    {...(activeItem && { 'aria-activedescendant': activeItem })}
+                    role="combobox"
+                    isExpanded={isOpen}
+                    aria-controls="select-typeahead-listbox"
+                    aria-label={textLabel}
+                />
+
+                <TextInputGroupUtilities>
+                    {!!value && (
+                        <Button
+                            variant="plain"
+                            onClick={() => {
+                                onChange('');
+                                setFilterValue('');
+                                textInputRef?.current?.focus();
+                            }}
+                            aria-label="Clear input value"
+                        >
+                            <TimesIcon aria-hidden />
+                        </Button>
+                    )}
+                </TextInputGroupUtilities>
+            </TextInputGroup>
+        </MenuToggle>
+    );
+
+    return (
+        <>
+            <Select
+                aria-label={`${textLabel} select menu`}
```

Comment:
Nit: could we use a static value here (and the other `aria-label` above) instead of string concatenation?

Reply:
What would the benefit be of making it static? The `textLabel` is used for these specific aria labels:

1. MenuToggle - "**Filter results by image name** menu toggle"
2. TextInputGroupMain - "**Filter results by image name**"
3. Select - "**Filter results by image name** select menu"

And depending on what entity/attribute pairing we choose, it will change

Reply:
The only benefit would be to aid in potential internationalization down the line, by avoiding string concatenation. Would something like `Filter results select menu` be roughly as valuable as having the entity/attribute as part of the text?

Reply:
Yea, that's reasonable. I'll do a follow up just to get this going